jobhub.go

```diff
@@ -72,47 +66,40 @@ var statusDescription = map[StatusCode]string{
 	Succeeded: "Succeeded",
 }
 
-func (jStatus JobStatus) LastExecutionStatus() *ExecutionStatus {
-	if len(jStatus.Statuses) > 0 {
-		return &jStatus.Statuses[len(jStatus.Statuses)-1]
+func (j JobStatus) LastExecutionStatus() *ExecutionStatus {
+	if len(j.Statuses) > 0 {
+		return &j.Statuses[len(j.Statuses)-1]
 	}
 	return nil
 }
 
-func (sCode StatusCode) String() string {
-	return statusDescription[sCode]
+func (s StatusCode) String() string {
+	return statusDescription[s]
 }
 
-func (j Job) String() string {
-	return fmt.Sprintf("Pipeline ID: %d | ID: %d | Name: %s", j.pipelineID, j.id, j.Name)
-}
-
-func NewPipeline() *Pipeline {
-	return &Pipeline{
-		Log:             logrus.StandardLogger(),
-		jobByID:         make(map[int]Job),
-		jobDependency:   make(map[int][]int),
-		startingJob:     make(map[int]bool),
-		recursionLevels: make(map[int]int),
+func NewPipeline(name string) *pipeline {
+	return &pipeline{
+		name:          name,
+		jobByID:       make(map[int]Job),
+		jobDependency: make(map[int][]int),
 	}
 }
 
 func nextIDPipeline() int {
-	tempID := nextPipelineID
-	nextPipelineID++
+	tempID := pipelineIDPool
```

Comment:
pool usually means that we have a limited number of some resource, like dababase connection pool or file descriptor pool, which needs to be shared across multiple users.

Reply:
Done; renamed to previous.